### ISSUE TITLE
Multiple entries for a rebus square

### ISSUE THREAD

**Issue by malihanda** [mdirolf/crosshare#310]
Sometimes rebuses (rebii??) are different when read in different directions. Would be nice to have either one count as "correct." See, e.g. the discussion [here](https://rosswordpuzzles.com/2021/05/16/let-it-burn/).

AcrossLite doesn't support this ([....Or does it??](https://twitter.com/litsoft/status/1423107193591656453?s=20)), so the "export to .puz" might get hacky.

**Comment by mdirolf** [comment]
Duplicate of #261 

I'm not too worried about the .puz thing - there are a couple of things that already don't play nicely with .puz (meta-puzzle solution/options, shade vs circle for cell highlights, etc.) - I'd likely just pick the first solution specified and export it as that.

I've mainly been waiting on this to think out the best way to represent it to allow future flexibility. Currently I'm thinking you'd have a normal grid and then be able to specify an arbitrary number of "alternates". Each alternate is an array of cell indexes w/ possible values (either a rebus or a single char).

So to add a rebus w/ multiple values you'd add a single alternate w/ a single cell w/ the other value.

To have multiple rebiiiiiii each w/ multiple values you'd add an alternate for each.

To do the classic CLINTON/BOBDOLE thing you'd have one of them in the grid as normal and add a single alternate w/ all 7 cells specified in the array.

I think this should work well and covers all the use cases I can think of - anything you can think of that that wouldn't support @malihanda?

Remaining questions:
- How to specify these in the constructor UI
- How to get these to play nicely w/ check/reveal, particularly in the multi-cell case (e.g. if you check BOXDOLE should it know that you only had the X wrong or should it just check against the default grid and tell you they're all wrong? Likely the latter to start and the former could be a follow-up issue)
- Perhaps as a followup issue, should we do the thing NYT does where the alternates flash back and forth when the puzzle is complete? Or some other way of showing the solver all the different alternates.

**Comment by mdirolf** [comment]
Yup they're definitely two different cases - I just think they're going to end up being similar in terms of UX and also in terms of needed changes to the underlying data format. So I'm hoping to solve them both with one implementation.

**Comment by mdirolf** [comment]
> In terms of "different cases that might end up being similar in terms of changes to data format"........... Do you have an open issue for allowing blank squares in puzzles?

Neat idea - just opened one!

**Comment by mdirolf** [comment]
This has been deployed, please LMK if there are any issues. Alternate solutions can be added from the constructor or the puzzle edit page.

Note that for the case you asked about originally you'd add a separate alternate solution for each rebus square - that way the solver can mix-and-match and still get it marked correct.
